src/stores/windowStore.ts

```diff
@@ -0,0 +1,358 @@
+import create from 'zustand';
+import { Vector2, Vector3 } from 'three';
+import { calculateTilePositions } from '../Utils/tileModes';
+import { Camera } from '@react-three/fiber';
+
+export interface WindowInf {
+  id: string;
+  title: string;
+  subtitle?: string;
+  icon?: string;
+  position: Vector3;
+  content: React.ReactNode;
+  width: number;
+  height: number;
+  isMinimized: boolean;
+  isFullscreen: boolean;
+  scale: Vector3;
+  isFocused?: boolean;
+  isClosable?: boolean;
+  isResizable?: boolean;
+  isMovable?: boolean;
+  onClose?: () => void;
+  onMinimize?: () => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  onClick?: () => void;
+  onHover?: () => void;
+  onHoverEnd?: () => void;
+  disableTitleBar?: boolean;
+  disableIcon?: boolean;
+  disableActionBtns?: boolean;
+  disableBackground?: boolean;
+  disableAdjustSize?: boolean;
+  opacity?: number; // Added opacity option
+}
+
+interface WindowStore {
+  windows: Record<string, WindowInf>;
```

Comment:
seems like the id is the key of the record and also inside the window
would use a array here instead

Reply:
It is a random uuid, that can be set right now. This would allow for static ID in case of removal of windows. In an array that identifier can change.